src/treecons.ml

```diff
@@ -0,0 +1,138 @@
+(** Tree constructors: a simpler representation of terms to be used by a
+    decision tree. *)
+
+open Extra
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  | Symb of
+      { c_sym : Files.module_path
+      (** Fully qualified name of the constructor. *)
```

Comment:
This is not what I expected. Isn't Files.module_path just the module part in Terms.sym?

Why not using c_sym : sym and sym_cmp to define compare?

Reply:
Ah, we have a sym_cmp?

Reply:
I can't use types from basics as it would create a cyclic dependency.  Or I'd have to refactor quite a lot of code.

Reply:
I see.

Reply:
Then, indeed, it was better to have 2 fields, one for the the module_path and one for the name like in Terms.sym. I suggested to use 1 field only because I thought we could use the type sym, but I didn't think of dependencies. So, could you please come back to 2 fields to preserve the semantics of module_path? Sorry for the inconvenience.

Reply:
Yes, there is a sym_cmp (added by you ;-)) but it is defined in Basics...

Reply:
fine fine